### ISSUE TITLE
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: DioError [DioErrorType.DEFAULT]: RangeError (index): Invalid value: Only valid value is 0: 1#0      List.[] (dart:core-patch/growable_array.dart:145:60)

### ISSUE THREAD

**Issue by Q14** [flutterchina/dio#303]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  ios    模拟器                  |      |
| Platform Version               | 12.0                               |      |
| Dio Version                    | 2.1.0                                 |      |
| Android Studio / Xcode Version | Android Studio 3.3.2            |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Unhandled Exception: DioError [DioErrorType.DEFAULT]: RangeError (index): Invalid value: Only valid value is 0: 1#0      List.[] (dart:core-patch/growable_array.dart:145:60)
#1      RequestOptions.uri (package:dio/src/options.dart:217:29)
#2      DefaultHttpClientAdapter.fetch (package:dio/src/adapter.dart:120:67)
<asynchronous suspension>
#3      Dio._makeRequest (package:dio/src/dio.dart:735:46)
<asynchronous suspension>
#4      Dio._request.<anonymous closure>.<anonymous closure> (package:dio/src/dio.dart:707:22)
#5      _rootRunUnary (dart:async/zone.dart:1132:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#7      _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#8      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#9      Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#10     Future._complete (dart:async/future_impl.dart:473:7)
#11     _SyncCompleter.complete (dart:async/<…>

**Comment by MonkeyInWind** [comment]
+1
dart写的接口   postman  web都可以调用

**Comment by MonkeyInWind** [comment]
找到原因了  android模拟器把宿主机ip指定为了10.0.2.2  所以不要访问电脑的ip  要访问10.0.2.2这个ip @Q14

**Comment by KentFeng** [comment]
大佬，这个问题怎么解决啊 @MonkeyInWind

**Comment by scofield-hello** [comment]
ios运行也有相同的问题，怎么解决呢

**Comment by dxjdqc** [comment]
> 找到原因了 android模拟器把宿主机ip指定为了10.0.2.2 所以不要访问电脑的ip 要访问10.0.2.2这个ip @Q14

请问下，我用的是真机测试也报这个错，怎么解决呢

**Comment by amreniouinnovent** [comment]
my base URL in dio was empty "" now solved

**Comment by wendux** [comment]
如果该问题还存在，请按模板补全信息，然后给出最小的可复现步骤后再重新打开


 :octocat: [From gitme Android](http://flutterchina.club/app/gm.html)

**Comment by hjJunior** [comment]
I got the sample error, but in my case it was because my base URL in dio was missing the `http` prefix
